src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -287,29 +336,235 @@ public TestResult findCorrespondingResult(String id) {
+     */
+    public void doGraphMapUtil (StaplerRequest req, StaplerResponse rsp) throws IOException {
+        String projectLevel = getParameter(req, PROJECTLEVEL);
+        String trendType = getParameter(req, TRENDTYPE);
+        int index = Arrays.binarySearch(projectList, projectLevel);
+        if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) && trendType.equals(BUILD_ANALYSIS)) {
+
+            /*
+             * This method generates a mapping from chart coordinates to url, to redirect to on clicking the
+             * trend generated by same conditions in {@link #doGraphUtil(StaplerRequest, StaplerResponse)}
+             */
+            ChartUtil.generateClickableMap(req, rsp, createChart(req, buildDataSetPerProject(req)), calcDefaultSize());
+        } else if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) && trendType.equals(LENGTHY_TESTS)) {
+
+            /*
+             * This method generates a mapping from chart coordinates to url, to redirect to on clicking the
+             * trend generated by same conditions in {@link #doGraphUtil(StaplerRequest, StaplerResponse)}
+             */
+            ChartUtil.generateClickableMap(req, rsp, createChart(req, buildLengthyTestDataset(req)), calcDefaultSize());
+        } else if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) && trendType.equals(FLAKY_TESTS)) {
+
+            /*
+             * This method generates a mapping from chart coordinates to url, to redirect to on clicking the
+             * trend generated by same conditions in {@link #doGraphUtil(StaplerRequest, StaplerResponse)}
+             */
+            ChartUtil.generateClickableMap(req, rsp, createXYChart(req, buildFlapperDataset(req)), calcDefaultSize());
+        } else {
+
+            /*
+             * This method is invoked when a user deliberately fires a wrong url with invalid query
+             * parameters and it generates a mapping for the same scenario as in
+             * {@link #doGraphUtil(StaplerRequest, StaplerResponse)}.
+             */
+            ChartUtil.generateClickableMap(req, rsp, createChart(req, buildDataSet(req)), calcDefaultSize());
+        }
+    }
+
+    /**
+     * A method to extract value of query parameters from url.
+     * @param req The HTTP request message.
+     * @param paramName The name of the query parameter to be extracted.
+     * @return The extracted value of the of the query parameter.
+     * <p>
+     * If the user deliberately fires url with less query parameters then those missing query parameters
+     * are assigned default values.
+     */
+    private String getParameter (StaplerRequest req, String paramName) {
+        String paramValue = req.getParameter(paramName);
+        if (paramValue == null) {
+
+            /*
+             * The default values for each of the mandatory query parameter.
+             */
+            if (paramName.equals(FAILUREONLY))
```

Comment:
use switch instead, that way you won't have to do NULL check for `paramValue`

Reply:
Done.